File: Problems/No.2202_Maximize_the_Topmost_Element_After_K_Moves/maximize_the_topmost_element_after_k_moves.py
class Solution(object):
    def maximumTop(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """

        # parameter
        max_value = 0

        # start computing
        if ((len(nums) == 1) and (k%2 == 1)):
            return -1
        else:
            for i in range(min(len(nums), k-1)):
                max_value = max(max_value, nums[i])
            if (k < len(nums)):
                max_value = max(max_value, nums[k])
            return max_value

File: Problems/No.2202_Maximize_the_Topmost_Element_After_K_Moves/maximize_the_topmost_element_after_k_moves.cpp
class Solution {
public:
    int maximumTop(vector<int>& nums, int k) {
        
        // parameter
        int max_value = 0;

        // start computing
        if ((nums.size() == 1) && (k%2 == 1)) return -1;
        else if (nums.size() < k) return *std::max_element(nums.begin(), nums.end());
        else {

            for (int i=0; i<k-1; i++) max_value = std::max(max_value, nums[i]);
            if (k < nums.size()) max_value = std::max(max_value, nums[k]);
            return max_value;

        }

    }
};
